src/app/services/auth.service.ts

```diff
@@ -57,21 +58,43 @@ export class AuthService {
   }
 
 
-  // isTokenExpired(token: string): boolean {
-  //   const decodedToken: any = jwtDecode(token)
-  //   return (decodedToken.exp * 1000) < Date.now()
-  // }
-
   private handleError(error: HttpErrorResponse) {
-    let errorMessage = 'An unknown error occurred';
+    let errorMessage: string;
+
     if (error.error instanceof ErrorEvent) {
-      // Client-side or network error
+    
       errorMessage = `Error: ${error.error.message}`;
     } else {
-      // Backend error
-      errorMessage = `Error Code: ${error.status}\nMessage: ${error.message}`;
+    
+      switch (error.status) {
+        case 400:
+
+        errorMessage = `Bad Request: ${error.error?.message || error.message}`   
+          break;
+
+          case 401: 
+        errorMessage  = 'unathorized: You need to log in to access this resource';  
+          break;
+
+          case 500: 
+        errorMessage = 'Internal Server Error: Please try again later';
+          break;
+          
+          case 503:
+        errorMessage = 'Sevice Unavailable: The server is temporarily unable to handle the request';
+           break;    
```

Comment:
Since these messages will be used else where, probably in another service, is it possible to have named constants let's say enum and assigned them here?

Reply:
done